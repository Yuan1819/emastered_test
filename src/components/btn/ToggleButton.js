import {useState} from "react";


export function ToggleButton(props) {
  const handleClick = () => {
    props.setToggleRef(!props.toggleRef);
  }
  return (
    <div id="normalReferenceToggle" className="can-toggle">
      <input id="toggleReferenceUpload" type="checkbox" checked={props.toggleRef} />
      <label htmlFor="toggleReferenceUpload" onClick={handleClick}>
        <div className="can-toggle__switch" data-checked="REFERENCE" data-unchecked="NORMAL"></div>
      </label>
    </div>
  )
}
