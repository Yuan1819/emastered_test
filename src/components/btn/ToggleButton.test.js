import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import renderer from 'react-test-renderer';
import {ToggleButton} from "./ToggleButton";


it("should render a ToggleButton", () => {
  let component = renderer.create(<ToggleButton/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const testInstance = component.root;
});
