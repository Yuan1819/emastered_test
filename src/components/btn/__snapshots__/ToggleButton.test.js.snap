// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a ToggleButton 1`] = `
<div
  className="can-toggle"
  id="normalReferenceToggle"
>
  <input
    id="toggleReferenceUpload"
    type="checkbox"
  />
  <label
    htmlFor="toggleReferenceUpload"
    onClick={[Function]}
  >
    <div
      className="can-toggle__switch"
      data-checked="REFERENCE"
      data-unchecked="NORMAL"
    />
  </label>
</div>
`;
